overlays/custom-packages/labwc/labwc-colored-borders.patch

```diff
@@ -0,0 +1,521 @@
++	if (strlen(hex) > 7) {
++		rgba[3] = atoi(hex + 7) / 100.0;
++	} else {
++		rgba[3] = 1.0;
++	}
++}
++
+ #endif /* LABWC_THEME_H */
+diff --git a/include/window-rules.h b/include/window-rules.h
+index fae1daf..ff8163e 100644
+--- a/include/window-rules.h
++++ b/include/window-rules.h
+@@ -18,6 +18,7 @@ enum property {
+  *   - 'app_id' for native Wayland windows
+  *   - 'WM_CLASS' for XWayland clients
+  */
++
+ struct window_rule {
+ 	char *identifier;
+ 	char *title;
+@@ -32,11 +33,27 @@ struct window_rule {
+ 	enum property ignore_focus_request;
+ 
+ 	struct wl_list link; /* struct rcxml.window_rules */
++
++	/* Customisation window title and borders*/
++	bool has_custom_border;
++	float custom_border_color[4];
+ };
+ 
+ struct view;
+ 
++void init_window_rule(struct window_rule *rule);
++
+ void window_rules_apply(struct view *view, enum window_rule_event event);
+ enum property window_rules_get_property(struct view *view, const char *property);
+ 
++/**
++ * window_rules_get_custom_border_color - check for presence of custom color in window rules
++ * @view: view data
++ * @color: pointer to color array, where custom color will be copied to
++ * @return: true if color was found or false if not
++ */
++bool window_rules_get_custom_border_color(struct view *view, float *rgba);
++
++
++
+ #endif /* LABWC_WINDOW_RULES_H */
+diff --git a/src/config/rcxml.c b/src/config/rcxml.c
+index c8da660..57e40f0 100644
+--- a/src/config/rcxml.c
++++ b/src/config/rcxml.c
+@@ -29,6 +29,7 @@
+ #include "regions.h"
+ #include "window-rules.h"
+ #include "workspaces.h"
++#include "theme.h"
+ 
+ static bool in_regions;
+ static bool in_usable_area_override;
+@@ -112,6 +113,7 @@ fill_window_rule(char *nodename, char *content)
+ {
+ 	if (!strcasecmp(nodename, "windowRule.windowRules")) {
+ 		current_window_rule = znew(*current_window_rule);
++		init_window_rule(current_window_rule);
+ 		wl_list_append(&rc.window_rules, &current_window_rule->link);
+ 		wl_list_init(&current_window_rule->actions);
+ 		return;
+@@ -127,6 +129,8 @@ fill_window_rule(char *nodename, char *content)
+ 	} else if (!strcmp(nodename, "identifier")) {
+ 		free(current_window_rule->identifier);
+ 		current_window_rule->identifier = xstrdup(content);
++		wlr_log(WLR_INFO, "Identifier found: %s=\"%s\"",
++                        nodename, content);
+ 	} else if (!strcmp(nodename, "title")) {
+ 		free(current_window_rule->title);
+ 		current_window_rule->title = xstrdup(content);
+@@ -153,6 +157,14 @@ fill_window_rule(char *nodename, char *content)
+ 	} else if (!strcasecmp(nodename, "ignoreFocusRequest")) {
+ 		set_property(content, &current_window_rule->ignore_focus_request);
+ 
++	/* Custom border properties: color */
++	} else if (!strcasecmp(nodename, "borderColor")) {
++		parse_hexstr(content, current_window_rule->custom_border_color);
++		current_window_rule->has_custom_border = true;
++		wlr_log(WLR_DEBUG, "Custom borderColor was found in config: %s, parsed into: %f, %f, %f, %f\n",
```

Comment:
The code indentation here on lines 174-176 is non-consistent which kind of adds to hackiness of the concept patch. Please indent.